js/chat_app/src/components/chatComponents/Chat.tsx

```diff
@@ -0,0 +1,162 @@
+import { Box, Container, TextField } from "@mui/material";
+import React, { useContext, useEffect, useState } from "react";
+import { UserContext } from "../UserContext";
+
+import Timeline from "@mui/lab/Timeline";
+import TimelineItem from "@mui/lab/TimelineItem";
+import TimelineSeparator from "@mui/lab/TimelineSeparator";
+import TimelineConnector from "@mui/lab/TimelineConnector";
+import TimelineContent from "@mui/lab/TimelineContent";
+import TimelineDot from "@mui/lab/TimelineDot";
+import TimelineOppositeContent from "@mui/lab/TimelineOppositeContent";
+
+interface dateOptions {
+  day: "numeric";
+  month: "numeric";
+  year: "numeric";
+  hour: "numeric";
+  minute: "numeric";
+  second: "numeric";
+}
+
+type messageListType = {
+  id: string;
+  message: string;
+  authorId: string;
+  date: number;
+};
+
+type usersListType = {
+  id?: string;
+  username?: string;
+};
+
+export const Chat: React.FC = () => {
+  const [messagesList, setMessagesList] = useState<messageListType[]>([]);
+  const [usersList, setUsersList] = useState<usersListType[]>([]);
+  const [message, setMessage] = useState<string>("");
+  const { auth } = useContext(UserContext);
+  const chatId: string = "e2f96143-b219-40e1-b6dc-a9a6a0cd01c0";
+
+  const options: dateOptions = {
+    day: "numeric",
+    month: "numeric",
+    year: "numeric",
+    hour: "numeric",
+    minute: "numeric",
+    second: "numeric",
+  };
+
+  useEffect(() => {
+    getUsersList();
+    const interval = setInterval(() => {
```

Comment:
Consider not sending the next request before handling the previous one.
Or, as an alternative - do not call the previous `setMessagesList` if a new request has been sent.